node/src/components/block_validator.rs

```diff
@@ -403,20 +378,14 @@ where
     I: Clone + Send + PartialEq + Eq + 'static,
 {
     let validate_deploy = move |result: FetchResult<Deploy, I>| match result {
-        FetchResult::FromStorage(deploy) | FetchResult::FromPeer(deploy, _) => {
-            if deploy
-                .header()
-                .is_valid(&chainspec.deploy_config, block_timestamp)
-            {
-                let deploy_size = deploy.serialized_length();
+        FetchResult::FromStorage(deploy) | FetchResult::FromPeer(deploy, _) => deploy
+            .deploy_type()
```

Comment:
Why would that even fail?

Reply:
These are the relevant lines from `Deploy::deploy_type`:
```rust
                let value = payment_item
                    .args()
                    .get(ARG_AMOUNT)
                    .ok_or(Error::InvalidPayment)?;
                let value = value
                    .clone()
                    .into_t::<U512>()
                    .map_err(|_| Error::InvalidPayment)?;
```